specs/agents/logging.md

```diff
@@ -39,3 +38,59 @@ Those agents should still accept their "native" log levels to preserve backwards
 However, in central config,
 there will only be a dropdown with the levels that are consistent across agents.
 Also, the documentation should not mention the old log levels going forward.
+
+### `log_ecs_formatting` configuration
+
+Configures the agent to automatically format logs as ECS-compatible JSON
+(if possible).
+
+|----------------|---|
+| Valid options  | `on`, `off` |
+| Default        | `off`  |
+| Dynamic        | `false` |
+| Central config | `false` |
+
+Not all agents will be able to automatically format logs in this way. Those
+agents should not implement this configuration option.
+
+For some agents, additional options makes sense. For example, the Java agent
+also accepts the values `shade` and `replace`.
+
+When this option is set to `on`, the agent should format all logs from the
+app as ECS-compatible json, as shown in the
+[spec](https://github.com/elastic/ecs-logging/blob/master/spec/spec.json).
+
+#### Required fields
+
+The following fields are required:
+
+* `@timestamp`
+* `log.level`
+* `message`
+* `ecs.version`
+
+#### Recommended fields
+
+The following fields are important for a good user experience in Kibana,
+but will not cause errors if they are omitted:
+
+##### `service.name`
+
+Agents should always populate
+[`service.name`](https://github.com/elastic/ecs-logging/blob/18cde109acb284c97988f9df9defb685b798db9a/spec/spec.json#L66-L74)
```

Comment:
While the line numbers stay consistent with this permalink, it might light to outdated information. Maybe just link to the master branch and don't include line numbers. It requires searching for `service.name` in that spec but I think that's fine.